File: projects/tox4j/src/test/java/im/tox/gui/events/MessageTextOnKey.scala
package im.tox.gui.events

import java.awt.event.{KeyAdapter, KeyEvent}

import im.tox.gui.MainView

final class MessageTextOnKey(toxGui: MainView) extends KeyAdapter {
  override def keyPressed(event: KeyEvent): Unit = {
    if (event.getKeyChar == '\n') {
      toxGui.sendButton.doClick()
    }
  }
}

File: projects/tox4j/src/test/java/im/tox/gui/events/MessageTextOnKey.java
package im.tox.gui.events;

import im.tox.gui.MainView;

import java.awt.event.KeyAdapter;
import java.awt.event.KeyEvent;

public class MessageTextOnKey extends KeyAdapter {
  private MainView toxGui;

  public MessageTextOnKey(MainView toxGui) {
    this.toxGui = toxGui;
  }

  @Override
  public void keyPressed(KeyEvent event) {
    if (event.getKeyChar() == '\n') {
      toxGui.sendButton.doClick();
    }
  }
}
